### ISSUE TITLE
port tabfile 

### ISSUE THREAD

**Issue by lengyijun** [vaaandark/dioxionary#21]
Do you have any plan to port tabfile as well ?

**Comment by vaaandark** [comment]
I don't know what tabfile is. Could you please tell me what it is? :D

**Comment by lengyijun** [comment]
tabfile is part of stardict tools, 
It is used to build a dictionary from a text file
It is quite useful to build custom dictionary for me

**Comment by vaaandark** [comment]
Thank you for your information.
I will study Mr. Hu Zheng’s stardict project and documentation, evaluate the difficulty of the tabfile project, and then decide whether to support it.

**Comment by vaaandark** [comment]
> Thank you for your information. I will study Mr. Hu Zheng’s stardict project and documentation, evaluate the difficulty of the tabfile project, and then decide whether to support it.

我将主要参考 https://stardict-4.sourceforge.net/HowToCreateDictionary

**Comment by lengyijun** [comment]
A problem of tabfile is tabfile depends on dictzip, which maybe unavailable for some os

**Comment by vaaandark** [comment]
> A problem of tabfile is that it depends on dictzip, which maybe unavailable for some os
> 
> Personally, I install dictzip by nix

I think dictzip is just an implementation of zip, so you can use other zip as alternative.

**Comment by vaaandark** [comment]
> > A problem of tabfile is that it depends on dictzip, which maybe unavailable for some os
> > Personally, I install dictzip by nix
> 
> I think dictzip is just an implementation of zip, so you can use other zip as alternative.

I checked the document of stardict and found that dictzip is a modified
zip version that will provide a table for randomly accessing compressed
data. As a result, just using usual zip tools such as gzip also works. If
your homemade dictionary isn't too big, I think it is okay to use gzip
as an alternative.

By the way, I just learned that stardict's compression format has a
special table. In this project, I haven't supported it, which means
dioxionay may not be a good stardict alternative. But who knows, today's
computer is faster and has more RAM. Maybe compressing isn't so important
for modern stardicts : )
